src/lib/awsElasticSearch.js

```diff
@@ -0,0 +1,222 @@
+
+      return aws4.sign(request, credentials);
+    }
+  }
+  return {
+    Connection: AmazonConnection,
+  };
+};
+
+const getClient = async () => new Client({
+  ...createAwsConnector(
+    {
+      accessKeyId: access_key,
+      secretAccessKey: secret_key,
+    },
+    'us-gov-west-1',
+  ),
+  node: uri,
+});
+/*
+  Create an index that can have searchable documents assigned.
+*/
+const createIndex = async (indexName, mockClient) => {
+  try {
+    // Initialize the client.
+    const client = mockClient || await getClient();
+
+    // Create index.
+    const res = await client.indices.create({
+      index: indexName,
+    });
+
+    logger.info(`AWS OpenSearch: Successfully created index ${indexName}`);
+    return res;
+  } catch (error) {
+    auditLogger.error(`AWS OpenSearch Error: Unable to create index '${indexName}': ${error.message}`);
+    throw error;
+  }
+};
+/*
+  Assign a searchable document to an index.
+*/
+const addIndexDocument = async (indexName, id, document, mockClient) => {
+  try {
+    // Initialize the client.
+    const client = mockClient || await getClient();
+
+    // Add a document to an index.
+    const res = await client.index({
+      index: indexName,
+      id,
+      body: document,
+      refresh: true, // triggers manual refresh.
+    });
+    logger.info(`AWS OpenSearch: Successfully added document to index ${indexName}`);
+    return res;
+  } catch (error) {
+    auditLogger.error(`AWS OpenSearch Error: Unable to add document to index '${indexName}': ${error.message}`);
+    throw error;
+  }
+};
+/*
+  Search index documents.
+*/
+const search = async (indexName, fields, query, mockClient) => {
+  try {
+    // Initialize the client.
+    const client = mockClient || await getClient();
+
+    // Create search body.
+    const body = {
+      // size: 20,
+      query: {
+        multi_match: {
+          query,
+          fields,
+        },
+      },
+    };
+
+    // Search an index.
+    const res = await client.search({
+      index: indexName,
+      body,
+    });
+    logger.info(`AWS OpenSearch: Successfully searched the index ${indexName} using query '${query}`);
+    return res.body.hits;
+  } catch (error) {
+    auditLogger.error(`AWS OpenSearch Error: Unable to search the index '${indexName}' with query '${query}': ${error.message}`);
+    throw error;
+  }
+};
+
+/*
+  Update index document.
+*/
+/* Notes: Body must be in the below format for update. */
+/*
+    const body = {
+      doc: {
+        specialist: 'Bruce Wayne', // We are updating the document field value specialist.
+      },
+    };
+*/
+const updateIndexDocument = async (indexName, id, body, mockClient) => {
+  try {
+    // Initialize the client.
+    const client = mockClient || await getClient();
+
+    // Update index document.
+    const res = await client.update({
+      index: indexName,
+      id,
+      body,
+      refresh: true, // triggers manual refresh.
+    });
```

Comment:
I wonder if we want to refresh in our actual execution, from what we read the elastic search defaults are to refresh about every second, seems like that would be sufficient